app/questionnaire/questionnaire_store_updater.py

```diff
@@ -36,26 +39,81 @@ def is_dirty(self):
     def update_answers(self, form):
         self._update_questionnaire_store_with_form_data(form.data)
 
+    def get_relationship_collectors_by_list_name(self, list_name: str):
+        return self._schema.get_relationship_collectors_by_list_name(list_name)
+
     def update_relationship_answer(self, form_data, list_item_id, to_list_item_id):
         relationship_answer_id = self._schema.get_relationship_answer_id_for_block(
             self._current_location.block_id
         )
-
         answer = self._answer_store.get_answer(relationship_answer_id)
-        if answer:
+        self._create_relationship_store_and_update_answer(
+            relationship_answer_id, answer, form_data, list_item_id, to_list_item_id
+        )
+
+    def _create_relationship_store_and_update_answer(
+        self, relationship_answer_id, answer, form_data, list_item_id, to_list_item_id
+    ):
+        try:
             relationship_store = RelationshipStore(answer.value)
-        else:
+        except AttributeError:
             relationship_store = RelationshipStore()
 
         relationship_answer = form_data.get(relationship_answer_id)
-
         relationship = Relationship(list_item_id, to_list_item_id, relationship_answer)
         relationship_store.add_or_update(relationship)
-
         self._answer_store.add_or_update(
             Answer(relationship_answer_id, relationship_store.serialise())
         )
 
+    def mark_relationship_question_incomplete(self, list_name: str) -> None:
```

Comment:
Not sure about the naming. What about something like `incomplete_relationships_for_list`?

Reply:
For me its not marking the relationship question incomplete which makes me feel like a flag, its removing the location if it there are relationships. Possibly remove_relationship_list_completed

Reply:
Good point both - I'll update the name